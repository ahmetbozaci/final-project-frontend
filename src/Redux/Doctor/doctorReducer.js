import {
  FETCH_DOCTOR_SUCCESS,
} from './doctorActions';

const initialState = {
  doctors: [],
};

const doctorReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_DOCTOR_SUCCESS:
      return {
        doctors: action.payload,
      };

    default: return state;
  }
};

export default doctorReducer;
