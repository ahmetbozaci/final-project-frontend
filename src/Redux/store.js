import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './rootReducer';
import fetchDoctors from './doctor/doctorActions';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger)),
);
store.dispatch(fetchDoctors());

export default store;
