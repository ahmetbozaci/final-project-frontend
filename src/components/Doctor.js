/* eslint-disable max-len */
// import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';

// import axios from 'axios';
import '../app.css';
import { Carousel } from 'react-bootstrap';
import { Link } from 'react-router-dom';
// import fetchDoctors from '../Redux/doctor/doctorActions';

// import { Slide } from 'react-slideshow-image';

const Doctor = () => {
  const doctor = useSelector((state) => state.doctorReducer);
  const { doctors } = doctor;
  // const dispatch = useDispatch();
  console.log(doctor);

  if (doctors.length === 0) {
    return (
      <h3>loading....</h3>
    );
  }

  return (
    <Carousel className="caro">
      {
        doctors.map((doc) => (
          <Carousel.Item interval={2000} key={doc.id} className="maincaro">
            <img
              className="docimg"
              src={doc.picture}
              alt="First slide"
            />
            <Carousel.Caption className="caption">
              <h3 className="docname">
                Dr .
                {doc.name}
              </h3>
              <button className="docbtn" type="button"><Link to="/appointment" className="createapp"> Create-Appointment </Link></button>

            </Carousel.Caption>
          </Carousel.Item>
        ))
      }

    </Carousel>
  );
};
export default Doctor;
