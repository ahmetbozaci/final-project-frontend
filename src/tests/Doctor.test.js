import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Doctor from '../components/Doctor';

it('Doctor renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><Doctor /></Router></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
