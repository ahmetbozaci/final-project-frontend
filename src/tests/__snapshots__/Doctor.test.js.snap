// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor renders correctly 1`] = `
<div
  className="caro doctorMargin carousel slide"
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="carousel-indicators"
  />
  <div
    className="carousel-inner"
  />
  <a
    className="carousel-control-prev"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="visually-hidden"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="visually-hidden"
    >
      Next
    </span>
  </a>
</div>
`;
